subprojects/core/src/main/java/org/gradle/deployment/internal/DefaultDeploymentRegistry.java

```diff
@@ -84,11 +151,28 @@ public void stop() {
             handles.clear();
             lock.unlock();
         }
+        pendingChangesManager.removeListener(this);
     }
 
     private void failIfStopped() {
         if (stopped) {
             throw new IllegalStateException("Cannot modify deployment handles once the registry has been stopped.");
         }
     }
+
+    private static class PendingChanges {
+        private int pendingChanges;
```

Comment:
I don't really follow why this isn't a simple boolean flag. Might it be simpler to track:
a) is a build in progress?
b) has a change been made since the last build started? (We'd clear this as soon as a build commences).